File: src/main/java/command/real/sound/GuildMusicManager.java
package command.real.sound;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager;

/**
 * Holder for both the player and a track scheduler for one guild.
 */
public class GuildMusicManager {
	/**
	 * Audio player for the guild.
	 */
	public final AudioPlayer player;
	/**
	 * Track scheduler for the player.
	 */
	public final TrackScheduler scheduler;
	/**
	 * Wrapper around AudioPlayer to use it as an AudioSendHandler.
	 */
	public final AudioPlayerSendHandler sendHandler;

	/**
	 * Creates a player and a track scheduler.
	 *
	 * @param manager Audio player manager to use for creating the player.
	 */
	public GuildMusicManager(AudioPlayerManager manager) {
		player = manager.createPlayer();
		scheduler = new TrackScheduler(player);
		sendHandler = new AudioPlayerSendHandler(player);
		player.addListener(scheduler);
	}
}

File: src/main/kotlin/it/discordbot/command/music/config/GuildMusicManager.kt
package it.discordbot.command.music.config

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager

class GuildMusicManager(manager: AudioPlayerManager) {

	var player: AudioPlayer = manager.createPlayer()

	var scheduler: TrackScheduler = TrackScheduler(player)

	var sendHandler: AudioPlayerSendHandler = AudioPlayerSendHandler(player)

	var commandChannelID: String? = null

	init {
		player.addListener(scheduler)
	}

}